app/components/graph-editor/drawable-element.ts

```diff
@@ -0,0 +1,275 @@
+// File: drawable-element.ts
+// Created by: CJ Dimaano
+// Date created: February 4, 2017
+
+
+import { FONT_SIZE } from "./defaults";
+import { GraphEditorCanvas, point, rect, size } from "./graph-editor-canvas";
+import { Drawable } from "./drawable";
+import { DrawableGraph } from "./drawable-graph";
+import {
+    PropertyChangedEventArgs,
+    PropertyChangedEventEmitter,
+    PropertyChangedEventListener
+} from "./events";
+
+
+/**
+ * DrawableStates  
+ *   The states in which a drawable can be.
+ */
+export enum DrawableStates {
+    Default,
+    Dragging,
+    Hovered
+};
+
+/**
+ * DrawableElement  
+ *   Common properties shared between drawable edges and nodes.
+ */
+export abstract class DrawableElement extends Drawable {
```

Comment:
I just think that abstract classes usually indicate that there is coupled functionality (since some is provided by the abstract part and some by the subclass) that should be decoupled through composition.